src/scanner.cc

```diff
@@ -53,13 +53,13 @@ using namespace std;
 /**
  * Print input and result information.
  */
-bool debug = false;
+static bool debug = false;
```

Comment:
So I know you want to port this to C, but `constexpr` might help the compiler here?

Reply:
CPP macros ftw :)